generator/transit_generator.cpp

```diff
@@ -43,20 +59,82 @@ string GetFileName(string const & filePath)
   return name;
 }
 
-/// \brief Reads from |root| (json) and serializes an array to |serializer|.
-/// \param handler is function which fixes up vector of |Item|(s) after deserialization from json
-/// but before serialization to mwm.
 template <class Item>
-void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer,
-                     function<void(vector<Item> &)> handler = nullptr)
+void DeserializeFromJson(my::Json const & root, string const & key, vector<Item> & items)
 {
-  vector<Item> items;
-
+  items.clear();
   DeserializerFromJson deserializer(root.get());
   deserializer(items, key.c_str());
+}
+
+void DeserializerGatesFromJson(my::Json const & root, string const & mwmDir, string const & countryId,
+                               vector<Gate> & gates)
+{
+  DeserializeFromJson(root, "gates", gates);
+
+  // Creating IndexRouter.
+  CountryFile const countryFile(countryId);
+  LocalCountryFile localCountryFile(mwmDir, countryFile, 0 /* version */);
+  localCountryFile.SyncWithDisk();
+  CHECK_EQUAL(localCountryFile.GetFiles(), MapOptions::MapWithCarRouting,
+              ("No correct mwm corresponding local country file:", localCountryFile,
+               ". Path:", my::JoinFoldersToPath(mwmDir, countryId + DATA_FILE_EXTENSION)));
+
+  Index index;
+  pair<MwmSet::MwmId, MwmSet::RegResult> const result = index.Register(localCountryFile);
+  CHECK_EQUAL(result.second, MwmSet::RegResult::Success, ());
+
+  Platform platform;
```

Comment:
Are you sure that you don't want to use existing platform singleton?

Reply:
I do want to use the singleton. Thanks.